Sources/GRPCNIOTransportCore/Client/Connection/GRPCChannel.swift

```diff
@@ -82,6 +86,12 @@ package final class GRPCChannel: ClientTransport {
     self.input = AsyncStream.makeStream()
     self.connector = connector
 
+    if let authority = config.http2.authority ?? resolver.authority {
+      self.authority = PercentEncoding.encodeAuthority(authority)
```

Comment:
Why does the percent-encoding happen at this level of the hierarchy rather than when it is first obtained?

Reply:
If it remains sometimes percent-encoded and sometimes not when stored as a property on objects I think that should be documented.